### ISSUE TITLE
Acrylic UI

### ISSUE THREAD

**Issue by DylanFPS** [Microsoft/WindowsTemplateStudio#1499]
* **Overview:** I think that Acrylic design should be added as a feature to Windows Template Studio. Integrating Acrylic alongside WTS is tedious and it being properly integrated to WTS would be a great help.
* **Links:** https://docs.microsoft.com/en-us/windows/uwp/design/style/acrylic

## System

* **VS Version:** 2017 - 15.4.3
* **WTS Wizard Version:** 1.4.17291.5
* **WTS Template Version:** 1.4.17291.5
* **Windows Build:** 16299

**Comment by crutkas** [comment]
Hi Dylan, issue with adding in Acrylic is where does it make sense quickly followed by does it make sense from a "template" standpoint? We are going through with the Fluent team and to validate our assumptions.

The other critical support item we need to factor in is supporting the current and prior release of Windows.  A large sum of items were added into Fall Creators Update so we have to be very careful on where items are added in.

If there are critical areas you feel it makes sense (NavView for instance) please list out areas so we can have a discussion.  For NavView, we actually have a clever solution with the UWP Community Toolkit which will solve this.

 - clint

**Comment by DylanFPS** [comment]
Adding acrylic to WTS would be helpful and would make sense from a template
POV as it would provide some code for the user to customise and change how
UWP is implemented, similar to how a lot of WTS components are implemented
(in my eyes)

Supporting previous versions of Windows wouldn’t be hard as one of the apps
that has been sent to our publishing network (not made with WTS) works with
Acrylic and is backwards compatible. Acrylic just disables itself if the
Acrylic API is unavailable.

Adding designer support on previous versions would be hard but can’t you
just add a minimum Windows version to the templates? So that developers
that want to use WTS to implement the new features can still look to the
future? I haven’t taken a look at the WTS code - I’ll do it tomorrow to see
how it handles templates at the moment.

From personal experience, implementing Acrylic into a WTS app is hard, I
have never managed to do it successfully and the closed I’ve got is it
working but one page messes with the acrylic opacity on the others. I’ve
just given up with Acrylic with WTS until there’s an easier way.

**TL;DR**: Add acrylic but only make it available on FCU and higher, at the
moment acrylic says no to WTS

Dylan P
HOO at Ultz

**Comment by crutkas** [comment]
Dylan, WTS is just generates the base template code.  We build the base that needs to be a bit generic intentionally.  Acrylic brush itself is one line of code using FCU if that is your min/target version.  That can be done by in XAML.  https://docs.microsoft.com/en-us/windows/uwp/design/style/acrylic 

We won't add acrylic in spots unless it makes sense.  100% would love to know where you feel it is lacking.  Once again, for NavView, we are will have a plan but where else do you feel it is needed?

**Comment by DylanFPS** [comment]
Well at the moment I’m trying to integrate it with Pivot using the Tabs
template, so that’s where I think it should be the most. Can’t really think
of anywhere else off the top of my head.

**Comment by crutkas** [comment]
so you want acrylic in the header area for pivot control?  correct?

**Comment by DylanFPS** [comment]
In the navigation area preferably. I'm using the pivot template in WTS

**Comment by crutkas** [comment]
so found a bug but this should be pretty straight forward.  Set the pivot background color to {StaticResource SystemControlAcrylicWindowBrush}

Here is more documentation on the variants of acrylic you could do as well.
https://docs.microsoft.com/en-us/windows/uwp/design/style/acrylic

From here, you may want to do some adjustments to margins of the pages and set their background color to something solid.  This step does make me think we should revisit this on how we handle margins here.
